### ISSUE TITLE
OrderedFloat comparison with f32/f64 is not total

### ISSUE THREAD

**Issue by orlp** [reem/rust-ordered-float#143]
Is it intended that `OrderedFloat(x) == y` where`x, y` are float types is not a total comparison? This only matters for NaN: `OrderedFloat(f32::NAN) == OrderedFloat(f32::NAN)` is true, but `OrderedFloat(f32::NAN) == f32::NAN` is false.

On another note, why is `PartialEq<Rhs=T> for OrderedFloat<T>` implemented, but the same does not hold true for `PartialOrd<Rhs=T> for OrderedFloat<T>`?

**Comment by mbrubeck** [comment]
> Is it intended that `OrderedFloat(x) == y` where `x, y` are float types is not a total comparison? This only matters for NaN: `OrderedFloat(f32::NAN) == OrderedFloat(f32::NAN)` is true, but `OrderedFloat(f32::NAN) == f32::NAN` is false.

No, I don’t think this is intended; as far as I know we just never thought about which result would be better.

> On another note, why is `PartialEq<Rhs=T> for OrderedFloat<T>` implemented, but the same does not hold true for `PartialOrd<Rhs=T> for OrderedFloat<T>`?

Also just an oversight, apparently.
